### ISSUE TITLE
Middleware expects proper bool after upgrade to 2.2.1

### ISSUE THREAD

**Issue by MikaelElkiaer** [containous/traefik#6853]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
I upgraded from Docker image 2.0 to 2.2.1.
I added the new CRDs and updated RBAC as per the migration guide.

### What did you expect to see?
I checked if my routers and services were properly added based on my IngressRoutes.

### What did you see instead?
The routers and services were not there.
My pods were outputting an error in my middleware, something along the lines of parseBool failed - expected t or f but found " (double quote).

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
traefik/traefik:2.2.1
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
This is the Middleware definition that worked in 2.0:
```yaml
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: securityheaders
  namespace: traefik

spec:
  headers:
    stsPreload: "true"
    stsIncludeSubdomains: "true"
    stsSeconds: 31536000    
    forceSTSHeader: "true" 
    frameDeny: "true"
    sslRedirect: "true"

```
Note: I am running Traefik in a k3s cluster, using FluxCD to apply resources.

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
(paste your output here)
```
## Fix/workaround
As expected, I got everything up and running again by removing the "s from my bools in the Middleware definition.

I see this as a breaking change and am a bit surprised nobody else has run into this.
The official docs also specify these values in quotes, e.g.: https://docs.traefik.io/middlewares/headers/#using-security-headers

**Comment by dtomcej** [comment]
may be related to https://github.com/containous/traefik/issues/6760

**Comment by traefiker** [comment]
Closed by #6928.
